python/pyarrow/_array.pyx

```diff
@@ -1570,6 +1589,33 @@ cdef class DictionaryArray(Array):
         result.init(c_result)
         return result
 
+cdef class StructArray(Array):
+    @staticmethod
+    def from_arrays(field_names, arrays):
+        cdef:
+            Array array
+            shared_ptr[CArray] c_array
+            vector[shared_ptr[CArray]] c_arrays
+            shared_ptr[CArray] c_result
+
+        length = len(arrays[0])
```

Comment:
Should probably check that `arrays` isn't empty and raise a more informative error message than the `IndexError` that would be thrown as is.